### PR TITLE
feat: add specs for c++ compilers

### PR DESCRIPTION
**Pull request by gavrilikhin-d**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the new behavior (if this is a feature change)?**
Completion for `clang++`, `c++` (same as `clang`) and `g++` (same as `gcc`)

### CODE REVIEW

**Comment by gavrilikhin-d** [comment]
Is it possible to override some options with `loadSpec`? I'd like to override `-std` to add suggestions for C++ standards (`c++11`, `gnu++11`, etc)

File: src/g++.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const completionSpec: Fig.Spec = {
+  name: "g++",
+  description: "The default C++ compiler for most linux distributions",
+  loadSpec: "gcc",
+};
+export default completionSpec;
```
Comment (fedeci) [inline src/g++.ts:6]:
```suggestion
import gcc from "./gcc"
const completionSpec: Fig.Spec = {
  ...gcc,
  name: "g++",
  description: "The default C++ compiler for most linux distributions",
};
export default completionSpec;
```

**Review by fedeci — CHANGES_REQUESTED** [review]
For specs that only change name and description I would encourage to use imports since those are resolved at build time and not at runtime

**Comment by mschrage** [comment]
> Is it possible to override some options with `loadSpec`? I'd like to override `-std` to add suggestions for C++ standards (`c++11`, `gnu++11`, etc)

Now that you are importing the other completion spec at build time, you should be able to overwrite suggestions.

**Comment by gavrilikhin-d** [comment]
> > Is it possible to override some options with `loadSpec`? I'd like to override `-std` to add suggestions for C++ standards (`c++11`, `gnu++11`, etc)
> 
> Now that you are importing the other completion spec at build time, you should be able to overwrite suggestions.

Yes, i added these. However, for some reason g++ suggestions don't show up :(

**Comment by gavrilikhin-d** [comment]
And, maybe, it's better to generate suggestions from clang's [LangStandards.def](https://github.com/llvm/llvm-project/blob/main/clang/include/clang/Basic/LangStandards.def)?

File: src/clang++.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          "gnu++03",
+          "gnu++11",
+          "gnu++14",
+          "gnu++17",
+          "gnu++20",
+          "gnu++2b",
+        ],
+      },
+      requiresSeparator: true,
+    },
```
[the remark this answers is not in the document]
Comment (gavrilikhin-d) [reply src/clang++.ts]: Done! However, i still see no suggestions for gcc and g++. May it be because they are aliases to clang?

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**